### ISSUE TITLE
"Test" airflow connection does not work

### ISSUE THREAD

**Issue by swasenius** [apache/airflow#39968]
### Apache Airflow version

2.9.1

### What happened?

After migration 2.5.3 -> 2.9.1 we noticed that the connectivity "Test" in Airflow UI does not work for JDBC connections. We do have AIRFLOW__CORE__TEST_CONNECTION enabled.
It will give an error on the the UI such as: "`The conn_id <random string> isn't defined`". 
The string here is always changing so it somehow does not pick up the connection_id right.

### What you think should happen instead?

Testing JDBC connectivity should work, if connection details are correct.

### How to reproduce

When testing JDBC connection in Airflow UI.

### Operating System

Debian 12

### Versions of Apache Airflow Providers

airflow@airflow-worker-0:/opt/airflow$ pip freeze |grep airflow-prov
apache-airflow-providers-amazon==7.4.1
apache-airflow-providers-celery==3.6.2
apache-airflow-providers-cncf-kubernetes==8.0.0
apache-airflow-providers-common-io==1.3.1
apache-airflow-providers-common-sql==1.12.0
apache-airflow-providers-docker==3.10.0
apache-airflow-providers-elasticsearch==5.3.4
apache-airflow-providers-fab==1.0.4
apache-airflow-providers-ftp==3.8.0
apache-airflow-providers-google==10.17.0
apache-airflow-providers-grpc==3.4.1
apache-airflow-providers-hashicorp==3.6.4
apache-airflow-providers-http==4.10.1
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-jdbc==4.2.2
apache-airflow-providers-microsoft-azure==10.0.0
apache-airflow-providers-microsoft-mssql==3.6.1
apache-airflow-providers-mysql==5.5.4
apache-airflow-providers-odbc==4.5.0
apache-airflow-providers-openlineage==1.7.0
apache-airflow-providers-postgres==5.10.2
apache-airflow-providers-redis==3.6.1
apache-airflow-providers-sendgrid==3.4.0
apache-airflow-providers-sftp==4.9.1
apache-airflow-providers-slack==8.6.2
apache-airflow-providers-smtp==1.6.1
apache-airflow-providers-snowflake==5.4.0
apache-airflow-providers-sqlite==3.7.1
apache-airflow-providers-ssh==3.10.1

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by RNHTTR** [comment]
It's worth noting that this was disabled due to a [security vulnerability](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#testing-connections).

**Comment by aritra24** [comment]
Had a quick look, My guess would be the connection url being put in might be malformed in some way, something like "jdbc:postgresql://host:port/database?properties" while what it want's is just the host:port form? because the logs have this error message `Invalid connection string: jdbc://jdbc:postgresql://admin@localhost%3A5432%2Fairflow/?__extra__=%7B%7D.` I believe the error message in the UI could probably be clearer?

**Comment by swasenius** [comment]
Here's an example:
jdbc:sqlserver://XXX.XXX.XX.XX:1433
and this looks like MS SQL server. User ie login is define in it's own field.

**Comment by aritra24** [comment]
Yes @swasenius , the field expects just the XXX.XXX.XX.XX:1433 part and appends the jdbc:// part on it's own. I assume for MSSQL we are expected to use the Microsoft SQL Server connection type.

**Comment by swasenius** [comment]
If I remove the first part and use XXX.XXX.XX.XX:1433, I get:
Class com.microsoft.sqlserver.jdbc.SQLServerDriver is not found

**Comment by pateash** [comment]
I can check this out, If no one else is picking this up.
Please assign this to me.

**Comment by pateash** [comment]
Couple of things I found,

1. We can't provide jdbc: as it is automatically provided. => we need to add better documentation in UI and may be add a better error message.

<img width="1127" alt="image" src="https://github.com/user-attachments/assets/c10bd8e3-0871-4b50-9cb1-540ae9c71e3f">

2.

**Comment by bhatiazzz** [comment]
Hi @pateash / @swasenius ,

I am also facing the same did we have any solution yet?

**Comment by potiuk** [comment]
@pateash We are unassigning you from this issue as part of our updated [assignment policy](https://github.com/apache/airflow/blob/main/contributing-docs/04_how_to_contribute.rst#contribute-code-changes).

This is not meant to discourage your contribution — quite the opposite! You are still very welcome to work on this issue and submit a PR for it. Simply comment that you are working on it and open a PR when ready.

We found that formal assignments were not working well, as they often prevented others from contributing when the assignee was not actively working on the issue.
